01-JS-DOM-APIs/javascript.js

```diff
@@ -0,0 +1,117 @@
+function show(){
+                  let obj = document.getElementById("hidden");
+                  if(obj.style.display == "block"){
+                        obj.style.display = "none";
+                        document.getElementById("buttonShow").value= "Show";
+                      }
+                  else{
+                        obj.style.display = "block";
+                        document.getElementById("buttonShow").value= "Hide";
+                      }
+                    }
+//5-6
+function buttonJokeWithoutPromise() {
+    let xhttp = new XMLHttpRequest();
+    xhttp.open("GET", "http://api.icndb.com/jokes/random", true);
+    xhttp.onreadystatechange = function(event) {
+    if (xhttp.readyState === XMLHttpRequest.DONE)
+    {
+      let response = JSON.parse(event.target.response);
+      let txt = response.value.joke;
+      document.getElementById("jokeHidden").innerHTML = response.value.joke;
+      document.getElementById("buttonjoke").value="Another joke";
+               }
+           }
+    xhttp.send();
+      }
+//7-9
+function buttonJokeWithPromise(){
+            let config = {
+              url: "http://api.icndb.com/jokes/random"
+                  };
+                        let xhttp = new XMLHttpRequest();
+                        function resolve(xhttp) {
+                            let response = JSON.parse(event.target.response);
+                             document.getElementById("jokeHidden2").innerHTML = response.value.joke;
+                             if (xhttp.readyState == XMLHttpRequest.DONE) {
+                               if (xhttp.responseText.value) {
+                                 document.getElementById("button2").innerHTML  = xhttp.responseText.value.joke;
+                               }
+                             }
+                         }
+                         function reject() {
+                           document.getElementById("all").style.backgroundColor = "red";
+                         }
+                         eventRequestReusable(config).then(resolve,reject);
+                        };
+
+function eventRequestReusable(object, resolve, reject) {
+              let promise = new Promise( function (resolve, reject) {
+              let xhttp = new XMLHttpRequest();
+                       xhttp.open("GET", object.url , true);
+                       xhttp.send();
+                       xhttp.onload = function () {
+                         if (this.status === 200) {
+                           resolve(xhttp);
+                         } else {
+                           reject(this.statusText);
+                         }
+                       };
+                       xhttp.onerror = function () {
+                        reject(this.statusText);
+                       };
+                     });
+                     return promise;
+                   };
+//9-11
+function configAjax (methodHttp, url, asyncronic){
+ this.methodHttp = methodHttp;
```

Comment:
Notice that by doing ``this.method = methodHttp`` you are assignin a methodHttp variable to the global scope, that could be really problematic, instead define a config object, assign the attributes and return it, furthermore you can define that objConfig with the needed attributes instead of doing *new configAjax* that way this function won't be needed anymore.